dawn/src/dawn/SIR/SIR.cpp

```diff
@@ -722,16 +723,20 @@ BuiltinTypeID sir::Value::typeToBuiltinTypeID(sir::Value::Kind type) {
 }
 
 std::string sir::Value::toString() const {
+  std::ostringstream out;
   DAWN_ASSERT(has_value());
   switch(type_) {
   case Kind::Boolean:
     return std::get<bool>(*value_) ? "true" : "false";
   case Kind::Integer:
     return std::to_string(std::get<int>(*value_));
   case Kind::Double:
-    return std::to_string(std::get<double>(*value_));
+    out << std::setprecision(std::numeric_limits<double>::digits10 + 1)
```

Comment:
just curiosity, is the +1 necessary?

Reply:
Apparently yes https://en.cppreference.com/w/cpp/io/manip/setprecision
to convert from scale to precision